### PR TITLE
Add utils to facilitate applying filters

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

When building custom match filter functions for the `tableWithFilter` or the `selectWithFilter`, we often do the same logic to compare the filter value with different types of data. So I extracted this logic in some util functions that we can reuse when needed.

### Potential Breaking Changes

- `defaultMatchFilter` function that was used for filtering list box was renamed to `defaultListBoxMatchFilter`.

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [x] The potential breaking changes are clearly identified
-   [x] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/filterBox/tests/FilterBoxSelectors.spec.ts [hunk 1]
```diff
@@ -38,4 +40,64 @@ describe('FilterBox', () => {
             });
         });
     });
+
+    describe('defaultListBoxMatchFilter', () => {
```
Comment (gdostie) [inline packages/react-vapor/src/components/filterBox/tests/FilterBoxSelectors.spec.ts]: This whole described was only moved from another file

**Comment by gdostie** [comment]
Real usage example:
```ts
match(filterValue: string, {displayName, id, members, realms}: GlobalGroupModel): boolean {
    return (
        FilterUtils.matchesString(filterValue, displayName) ||
        FilterUtils.matchesArrayLength(filterValue, members) ||
        FilterUtils.matchesArrayLength(filterValue, realms) ||
        filterValue === id
    );
}
```

File: packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+import {IItemBoxProps} from '../itemBox/ItemBox';
 import {IFilterState} from './FilterBoxReducers';
 
 export interface GetFilterTextProps {
     id: string;
 }
 
+export type MatchFilter = (filterValue: string, item: IItemBoxProps) => boolean;
+
+export const defaultListBoxMatchFilter = (filterValue: string, item: IItemBoxProps) => {
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: Why did you move it in the selector? It make more sense in the Utils, no?

Comment (gdostie) [reply packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: Because its bound to `IItemBoxProps`, so it belongs somewhere in the `ItemBox` related files. I did not want it to be mistaken for the defaultMatchFilter of the table, which doesn't use `IItemBoxProps`. 
The selectors might not be the right file tho.

Comment (pochretien) [reply packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: I think we can scope these filter with something like 
```
FilterUtils. defaultMatchFilter
TableUtils.defaultMatchFilter
```
what do you think ?

Comment (GermainBergeron) [reply packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: I don't mind the name, but the location of the code seems really unusual to me

Comment (gdostie) [reply packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: FilterUtils are also relevant for the table

Comment (gdostie) [reply packages/react-vapor/src/components/filterBox/FilterBoxSelectors.ts]: Moved the defaultListBoxFilter to a new FilterBoxUtils file
